### PR TITLE
XGBoost: Fix MOJO loading when interaction_constraints is used

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-genmodel/src/main/java/hex/genmodel/attributes/ModelJsonReader.java [hunk 1]
```diff
@@ -314,6 +314,17 @@ private static Object convertBasedOnJsonType(final JsonElement convertFrom, fina
                         }
                         convertTo = arrS;
                         break;
+                    case STRING_ARR_ARR:
+                        final String[][] arrSS = new String[array.size()][];
+                        for (int i = 0; i < array.size(); i++) {
+                            final JsonArray arr2 = array.get(i).getAsJsonArray();
+                            arrSS[i] = new String[arr2.size()];
+                            for (int j = 0; j < array.size(); j++) {
```
Comment (mn-mikke) [inline h2o-genmodel/src/main/java/hex/genmodel/attributes/ModelJsonReader.java:322]: Shouldn't the for loop condition be `j < arr2.size()`?

**Review by Pscheidl — APPROVED** [review]
LGTM completely

File: h2o-genmodel/src/main/java/hex/genmodel/attributes/ModelJsonReader.java (hunk 1 above)
Comment (honzasterba) [reply h2o-genmodel/src/main/java/hex/genmodel/attributes/ModelJsonReader.java:322]: missing unit test

Comment (michalkurka) [reply h2o-genmodel/src/main/java/hex/genmodel/attributes/ModelJsonReader.java]: thx, test improved to check for expected output

**Review by honzasterba — APPROVED** [review]

**Comment by michalkurka** [comment]
@mn-mikke thank you for catching the bug - the test worked only thanks to a coincidence

thanks for the review and pointers @honzasterba and @Pscheidl

**Pull request merged**